core/imds/src/main/java/software/amazon/awssdk/imds/Ec2MetadataClientBuilder.java

```diff
@@ -63,7 +63,7 @@ public interface Ec2MetadataClientBuilder<B, T> extends SdkBuilder<Ec2MetadataCl
      * Define the endpoint of IMDS.
      * <p>
      * If not specified, the IMDS client will attempt to automatically resolve the endpoint value
-     * based on the logic of {@link Ec2MetadataEndpointProvider}.
+     * based on the logic of {@link Ec2MetadataConfigProvider}.
```

Comment:
Can we remove the mention of this internal API, everything after `resolve the endpoint value`?

Reply:
The reason for removal is we do not want to expose internal classes anywhere right ?